python/datasources/cdc_hiv.py

```diff
@@ -0,0 +1,228 @@
+
+                float_cols = [std_col.HIV_POPULATION_PCT,
+                              std_col.HIV_CASES,
+                              std_col.HIV_PER_100K,
+                              std_col.HIV_PCT_SHARE
+                              ]
+
+                column_types = gcs_to_bq_util.get_bq_column_types(
+                    df, float_cols=float_cols)
+
+                gcs_to_bq_util.add_df_to_bq(
+                    df, dataset, table_name, column_types=column_types)
+
+    def generate_breakdown_df(self, breakdown, geo_level):
+        COL_NAME = std_col.COUNTY_NAME_COL if geo_level == COUNTY_LEVEL else std_col.STATE_NAME_COL
+        FIPS = std_col.COUNTY_FIPS_COL if geo_level == COUNTY_LEVEL else std_col.STATE_FIPS_COL
+
+        cols_std = {
+            'Geography': COL_NAME,
+            'FIPS': FIPS,
+            'Age Group': std_col.AGE_COL,
+            'Sex': std_col.SEX_COL,
+            'Race/Ethnicity': std_col.RACE_OR_HISPANIC_COL,
+            'Year': std_col.TIME_PERIOD_COL,
+            'Cases': std_col.HIV_CASES,
+            'Rate per 100000': std_col.HIV_PER_100K,
+            'Population': std_col.POPULATION_COL,
+        }
+
+        cols = [
+            COL_NAME,
+            FIPS,
+            std_col.TIME_PERIOD_COL,
+            breakdown,
+            std_col.HIV_CASES,
+            std_col.HIV_PER_100K,
+            std_col.HIV_PCT_SHARE,
+            std_col.HIV_POPULATION_PCT]
+
+        source_dfs = []
+        missing_data = ['Data suppressed', 'Data not available']
+
+        # county fips needs 5 digits, state digits use 2
+        format_num = 5 if geo_level == COUNTY_LEVEL else 2
+        geo = COUNTY_LEVEL if geo_level == COUNTY_LEVEL else STATE_LEVEL
+
+        for group in GROUP_DICT[breakdown].keys():
+            source_group_df = gcs_to_bq_util.load_csv_as_df_from_data_dir(
+                f'cdc_hiv/{geo}', f'{breakdown}_{demo_dict.get(group, group)}_{geo}_{SOURCE_YEAR}.csv',
+                skiprows=9, thousands=',', dtype={'FIPS': str})
+            # adds leading zeros to fips
+            source_group_df['FIPS'] = source_group_df['FIPS'].str.zfill(
+                format_num)
+            source_dfs.append(source_group_df)
+
+        if geo_level != NATIONAL_LEVEL:
+            alls_df = generate_alls_df(geo_level)
+            source_dfs.append(alls_df)
+
+        merged_df = pd.concat(source_dfs, axis=0)
+        df = merged_df.rename(columns=cols_std)
+        df = df.sort_values([FIPS, breakdown]).reset_index(drop=True)
+
+        if geo_level == COUNTY_LEVEL:
+            df = merge_utils.merge_county_names(df)
+            df[std_col.STATE_FIPS_COL] = df[std_col.COUNTY_FIPS_COL].str.slice(
+                0, 2)
+
+        if geo_level == NATIONAL_LEVEL:
+            alls_df = generate_alls_df(NATIONAL_LEVEL)
+            nat_breakdown_df = generate_nat_from_state_df(breakdown)
+
+            merged_nat_df = pd.concat([nat_breakdown_df, alls_df])
```

Comment:
Also don't use `merged` in the name here. It implies it was created with a `pd.merge` IMO